### PR TITLE
REF: simplify should_extension_dispatch, remove dispatch_to_extension_op

### PR DESCRIPTION
**Pull request by jbrockmendel**
dispatch_to_extension_op was needed back before we got rid of integer-addition for DTA/TDA/Timestamp/Timedelta.  Now it reduces to a one-liner, which this inlines.

This also consolidates all of the casting/checks up-front, which in turn avoids a runtime import and simplifies the check for should_extension_dispatch.

### CODE REVIEW

File: pandas/core/ops/array_ops.py [hunk 1]
```diff
@@ -453,3 +442,87 @@ def get_array_op(op, str_rep: Optional[str] = None):
...
+    out : object
+
+    Notes
+    -----
+    Be careful to call this *after* determining the `name` attribute to be
+    attached to the result of the arithmetic operation.
+    """
+    from pandas.core.arrays import DatetimeArray, TimedeltaArray
+
+    if type(obj) is timedelta:
```
Comment (jreback) [inline pandas/core/ops/array_ops.py]: isinstance?

**Review by jreback — COMMENTED** [review]
looks good. small comment

File: pandas/core/ops/array_ops.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/ops/array_ops.py]: we used type to exclude Timedelta; this func is moved unchanged from `__init__` since it is only used in this module

**Comment by jreback** [comment]
thanks

**Pull request merged**